php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -968,6 +969,11 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         }
 
         $candID    = $_REQUEST['candID'] ?? '';
+
+        if (empty($candID)) {
```

Comment:
I think the issue was CommentID. 
You could add the check at the top of the function to return an empty array where CommentID is null.